### PR TITLE
Benchmark cases and compare.py script tool added

### PR DESCRIPTION
**Pull request by hidalgoss**
Please, review changes.

### CODE REVIEW

File: analysis/bm.out [hunk 1]
```diff
@@ -0,0 +1,12640 @@
+{
```
Comment (pablogh-2000) [inline analysis/bm.out:1]: remove this file

File: benchmark/benchmark_json/src/kpsr.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/kpsr.json:1]: is this file used at all? If not, remove it.

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 3]
```diff
@@ -0,0 +1,525 @@
...
+{
+    try{
+        // Create a json_string related to test conditions.
+        int i_vec_size = state.range(0);
+        std::string json_string;
+        size_t retrieved_vector_size=0;
+
+        myVectorFloats vec_f(i_vec_size);                                           // Load vector randomly with specified size.
+        std::string json_str = vec_f.serialize__Vector_to__JSON();                  // Now, use this vector to create a JSON serialization memory stream.
+#ifdef SAVE_FLOATS_TO_TEST_FILE                                                     // WARNING: Precompiler variable created through root CMakeLists file.
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:161]: does not make sense to use definition variables. Just include this part always or never.

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 4]
```diff
@@ -0,0 +1,525 @@
...
+float get_random_float_using_whole_range(){
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    return std::uniform_real_distribution<float>(std::numeric_limits<float>::max(), std::numeric_limits<float>::min())(gen);
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// cereal needs to know which data members to serialize in your classes. 
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+struct myVectorFloats {
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:59]: Use standard naming convention starting with upper case.
The name is not expressive, use `VectorFloatStruct`
Move this class to its own file

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 5]
```diff
@@ -0,0 +1,525 @@
...
+    std::vector<float> deserialize__JSON_file_to__Vector(const std::string & filename_string){
+        std::ifstream is(filename_string);
+        {
+            cereal::JSONInputArchive archive_in(is);
+            serialize(archive_in);
+        }
+        return _vec_f;
+    }
+
+    void load_random(int i_vec_size){
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:107]: rename to `fillWithRandomData`

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 6]
```diff
@@ -0,0 +1,525 @@
...
+        for(int i = 0; i < i_vec_size; i++) 
+        {
+            float f_rand_value = get_random_float_using_whole_range();
+            _vec_f.push_back(f_rand_value);
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:111]: avoid `push_back`. Better resize before the loop and use `[i]` access.
LOOK AT OTHER PLACES WITH THE SAME SITUATION

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 7]
```diff
@@ -0,0 +1,525 @@
...
+#include <fstream>  // ofstream
+
+#include <cereal/archives/json.hpp>
+#include <cereal/cereal.hpp>
+#include <cereal/types/vector.hpp>
+
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+bool save_to_myfile(const std::string & str, const std::string & fileName)
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:37]: Move all static methods to a new class with static methods in a class.
Call it `SerializationBenchmarkUtils`
Move it to its own file

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 8]
```diff
@@ -0,0 +1,525 @@
...
+        // Create a json_string related to test conditions.
+        int i_vec_size = state.range(0);
+        std::string json_string;
+        size_t retrieved_vector_size=0;
+
+        myVectorFloats vec_f(i_vec_size);                                           // Load vector randomly with specified size.
+        // TODO. Si no existe el archivo, lo creamos.
+        std::string filename = {JSON_FILE_PATH_AND_NAME + std::to_string(i_vec_size)};  // WARNING: Precompiler variable created through root CMakeLists file.
+        vec_f.serialize__Vector_to__JSON_file(filename);                            // Now, use this vector to create a JSON serialization disk file.
+        myVectorFloats vec_f_results;                                               // Create an empty vector to load results;
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:198]: rename `vec_f_results` to `deserializedVector`. Please be more careful with the naming of things.

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 9]
```diff
@@ -0,0 +1,525 @@
...
+        int i_vec_size                      = state.range(0);
+        int desired_Precission              = 17; //state.range(1);
+        bool std_fixed                      = true; //state.range(2);
+        std::string filename                = {JSON_FILE_PATH_AND_NAME + std::to_string(i_vec_size)};
+        simdjson::padded_string json_string = simdjson::padded_string::load(filename);
+
+        size_t retrieved_vector_size=0;
+        // This code gets timed
+        for (auto _ : state) {
+            simdjson::ondemand::parser parser;
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:231]: can the `parser` be placed outside the loop?

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 10]
```diff
@@ -0,0 +1,525 @@
...
+        int i_vec_size = state.range(0); // 3
+        int desired_Precission = 17; //state.range(1); //17
+        bool std_fixed         = true; //state.range(2);
+        std::string filename = {JSON_FILE_PATH_AND_NAME + std::to_string(i_vec_size)};  // WARNING: Precompiler variable created through root CMakeLists file.
+        simdjson::padded_string json_string = simdjson::padded_string::load(filename);
+        std::vector<float> vec_f(i_vec_size);
+
+        // This code gets timed
+        for (auto _ : state) {
+            simdjson::ondemand::parser parser;
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:277]: Can the `parser` be moved outside the loop?

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 11]
```diff
@@ -0,0 +1,525 @@
...
+    try{
+        // Create a json_string related to test conditions.
+        int i_vec_size          = state.range(0);
+        int desired_Precission  = 17; //state.range(1);
+        bool std_fixed          = true; //state.range(2);
+        std::string filename    = {JSON_FILE_PATH_AND_NAME + std::to_string(i_vec_size)};      // WARNING: Precompiler variable created through root CMakeLists file.
+        size_t retrieved_vector_size = 0;
+        // This code gets timed
+        for (auto _ : state) {
+            simdjson::ondemand::parser parser;
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:318]: Same as before

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 12]
```diff
@@ -0,0 +1,525 @@
...
+        bool std_fixed          = true; //state.range(2);
+        std::string filename    = {JSON_FILE_PATH_AND_NAME + std::to_string(i_vec_size)};      // WARNING: Precompiler variable created through root CMakeLists file.
+        // This code gets timed
+        for (auto _ : state) {
+            simdjson::ondemand::parser parser;
+            simdjson::padded_string json_string = simdjson::padded_string::load(filename);  // Now, we benchmarks spent time loading disk file.
+            auto doc = parser.iterate(json_string);
+
+            simdjson::ondemand::array jsonvec_f = doc.find_field("vector").get_array();
+            // Well-known vector size thus, do not use simdjson count_elements.
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:363]: remove comments that are irrelevant.

File: benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp [hunk 13]
```diff
@@ -0,0 +1,525 @@
...
+        // Create a json_string related to test conditions.
+        int i_vec_size                      = state.range(0); // 3
+        int desired_Precission              = 17; //state.range(1); //17
+        bool std_fixed                      = true; //state.range(2);
+        std::string filename                = {JSON_FILE_PATH_AND_NAME + std::to_string(i_vec_size)};      // WARNING: Precompiler variable created through root CMakeLists file.
+        simdjson::padded_string json_string = simdjson::padded_string::load(filename);
+
+        // This code gets timed
+        for (auto _ : state) {
+            simdjson::ondemand::parser parser;
```
Comment (pablogh-2000) [inline benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp:402]: same as before

**Comment by tomasz3dk** [comment]
Not required anymore

**Pull request closed without merging**